edge/services/helloworld/CreateService.md

```diff
@@ -0,0 +1,164 @@
+  - Uncheck **Securely store Docker logins in macOS keychain**
+
+3. Get a docker hub id at https://hub.docker.com/ , if you don't already have one. Log in to Docker Hub using your Docker Hub ID:
+
+  ```bash
+  export DOCKER_HUB_ID="<dockerhubid>"
+  echo "<dockerhubpassword>" | docker login -u $DOCKER_HUB_ID --password-stdin
+  ```
+
+  Output example:
+
+  ```bash
+  WARNING! Your password will be stored unencrypted in /home/pi/.docker/config.json.
+  Configure a credential helper to remove this warning. See
+  https://docs.docker.com/engine/reference/commandline/login/#credentials-store
+
+  Login Succeeded
+  ```
+
+4. Create a cryptographic signing key pair. This enables you to sign services when publishing them to the exchange. **This step only needs to be done once.**
+
+  ```bash
+  hzn key create "<x509-org>" "<x509-cn>"
+  ```
+
+  Where `<x509-org>` is your company name, and `<x509-cn>` is typically set to your email address.
+
+5. Install `git` and `jq`:
+
+  On **Linux**:
+
+  ```bash
+  sudo apt install -y git jq
+  ```
+
+  On **macOS**:
+
+  ```bash
+  brew install git jq
+  ```
+
+## <a id=build-publish-your-hw> Building and Publishing Your Own Hello World Example Edge Service
+
+1. `cd` to the directory in which you want to create your new service and then run this command to create the files for a simple edge service and associated Horizon metadata files:
+
+  ```bash
+  hzn dev service new -s myhelloworld -i "$DOCKER_HUB_ID/myhelloworld"
+  ```
+
+  Notice that some project variables are defined in `horizon/hzn.json` and referenced in other files, for example `horizon/service.definition.json`.
+
+2. Edit `service.sh` and change something simple, for example change "Hello" to "Hey there"
+
+  Note: this service is a shell script for brevity, but you can write your service in any language.
+
+3. Build the service docker image:
+
+  ```bash
+  make
+  ```
+
+4. Test the service by running it the simulated agent environment:
+
+  ```bash
+  hzn dev service start -S
+  ```
+
+5. Check that the container is running:
+
+  ```bash
+  sudo docker ps
+  ```
+
+6. See the environment variables Horizon passes into your service container:
+
+  ```bash
+  sudo docker inspect $(sudo docker ps -q --filter name=myhelloworld) | jq '.[0].Config.Env'
+  ```
+
+7. See the docker container running and look at the output:
+
+  on **Linux**:
+
+  ```bash
+  sudo tail -f /var/log/syslog | grep myhelloworld[[]
+  ```
+
+  on **Mac**:
+
+  ```bash
+  sudo docker logs -f $(sudo docker ps -q --filter name=myhelloworld)
+  ```
+
+8. Stop the service:
+
+  ```bash
+  hzn dev service stop
+  ```
+
+9. Have Horizon push your docker image to your registry and publish your service in the Horizon Exchange and see it there:
+
+  ```bash
+  hzn exchange service publish -f horizon/service.definition.json
+  hzn exchange service list
+  ```
+
+10. Publish your edge node deployment pattern in the Horizon Exchange and see it there:
```

Comment:
Would this sentence mean the same thing: Publish and view your edge node deployment pattern in the Horizon Exchange?

I ask, because using works like "see" and "Have" in step 9 are going to be problematic style-wise. I have to look at technically and stylistically satisfactory replacements.

Reply:
"Publish and view your edge node deployment pattern in the Horizon Exchange" is equivalent, so i made that change.